test/gamma.jl

```diff
@@ -80,6 +80,17 @@
         @test_throws MethodError logfactorial(1.0)
     end
 
+    @testset "BigFloat" begin
+        loggamma_mpfr(x::BigFloat) = invoke(SpecialFunctions._loggamma, Tuple{BigFloat}, x)
+        logabsgamma_mpfr(x::BigFloat) = invoke(SpecialFunctions._logabsgamma, Tuple{BigFloat}, x)
+        @test loggamma(big(3124)) == loggamma_mpfr(big(3124.0))
```

Comment:
I'm not sure what this is testing — that Julia is dispatching correctly?

Maybe test some known result, e.g. hard-coding a value from WolframAlpha?

Reply:
I added it to ensure that the results are exactly the same as the ones obtained with the `BigFloat` implementation directly, which indicates (but does not guarantee) that the arguments are promoted correctly. In https://github.com/JuliaMath/SpecialFunctions.jl/pull/349 I noticed that promotion was not working correctly for `beta` and hence I thought this would be useful.

I'll simplify and restructure it a bit and add some hard-coded values as well.

Reply:
I added more and better tests.